patches/minecraft/net/minecraft/entity/player/EntityPlayerMP.java.patch

```diff
@@ -120,13 +120,18 @@
          this.field_71070_bA = this.field_71069_bz;
      }
  
-@@ -1144,6 +1162,18 @@
+@@ -1144,6 +1162,23 @@
          this.field_193110_cw = p_193104_1_.field_193110_cw;
          this.func_192029_h(p_193104_1_.func_192023_dk());
          this.func_192031_i(p_193104_1_.func_192025_dl());
 +
 +        this.spawnChunkMap = p_193104_1_.spawnChunkMap;
 +        this.spawnForcedMap = p_193104_1_.spawnForcedMap;
++        if(p_193104_1_.field_71093_bK != 0)
```

Comment:
Remove this check in case the default spawn dimension is not overworld.

Reply:
That check is necessary though.
The player's respawn point (i.e. `EntityPlayer#getBedLocation(dim)`) is  copied before ``EntityPlayerMP#copyFrom`` and then set again *after* `EntityPlayerMP#copyFrom` in `PlayerList#recreatePlayerEntity`. If a bed in the overworld is broken `EntityPlayer#getBedSpawnLocation` will return `null` and thus the overworld spawn point will not be copied.
So the reason why this check must be there is that when a bed is destroyed in the overworld and the player respawns in the overworld it should not copy the respawn point, otherwise the "Your home bed was missing or obstructed" would show up every time you die instead of only once.